### PR TITLE
Add docs with current backend

### PR DESCRIPTION
**Pull request by tmigot**
update docstring and examples with the current backend system.

I also added a sentence on `nequ` as I remember students were struggling with it last year.

### CODE REVIEW

File: docs/src/index.md [hunk 1]
```diff
@@ -2,8 +2,8 @@
 
 This package provides AD-based model implementations that conform to the [NLPModels](https://github.com/JuliaSmoothOptimizers/ADNLPModels.jl) API. The following packages are supported:
 - `ForwardDiff.jl`: default choice.
-- `Zygote.jl`: you must load `Zygote.jl` separately and pass `ADNLPModels.ZygoteAD()` as the `adbackend` keyword argument to the `ADNLPModel` or `ADNLSModel` constructor.
-- `ReverseDiff.jl`: you must load `ReverseDiff.jl` separately and pass `ADNLPModels.ReverseDiffAD()` as the `adbackend` keyword argument to the `ADNLPModel` or `ADNLSModel` constructor.
+- `Zygote.jl`: you must load `Zygote.jl` separately and pass `ADNLPModels.ZygoteAD` as the `adbackend` keyword argument to the `ADNLPModel` or `ADNLSModel` constructor.
+- `ReverseDiff.jl`: you must load `ReverseDiff.jl` separately and pass `ADNLPModels.ReverseDiffAD` as the `adbackend` keyword argument to the `ADNLPModel` or `ADNLSModel` constructor.
```
Comment (dpo) [inline docs/src/index.md:6]: Is there doc explaining the arguments to the AD backends?

File: src/nlp.jl [hunk 2]
```diff
@@ -34,6 +34,46 @@ The following keyword arguments are available to the constructors for constraine
...
+# Examples
+```julia
+using ADNLPModels
+f(x) = sum(x)
+x0 = ones(3)
+nvar = 3
+ADNLPModel(f, x0) # uses the default ForwardDiffAD backend.
+
+using ReverseDiff
+ADNLPModel(f, x0; adbackend = ADNLPModels.ReverseDiffAD(nvar, f, x0))
```
Comment (dpo) [inline src/nlp.jl:55]: Maybe should revise the constructor before documenting this? It's pretty inconveninent.

Comment (tmigot) [reply src/nlp.jl:55]: Agreed. I can do a separate PR for this.

File: docs/src/index.md (hunk 1 above)
Comment (tmigot) [reply docs/src/index.md:6]: I will add docstrings for the backend as well then. There are the examples in the `ADNLPModel` docstring.

**Pull request merged**